### PR TITLE
Global tags

### PR DESCRIPTION
**Pull request by intelekshual**
This PR adds the ability to set global tags on a `Tracer` instance. These tags are applied to every `Span` created by the `Tracer`. This is useful when instrumenting applications in different environments (ex. staging and production), which is why I also updated the Rails contrib to set the `env` tag to the Rails environment.

### CODE REVIEW

**Comment by ufoot** [comment]
Thanks for the submission, will take a look, interesting!

**Comment by palazzem** [comment]
Hello @intelekshual ! we're looking at your PR and I'm going to rebase / merge it soon. Will keep you updated when it will be part of the next release!

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -64,7 +64,8 @@ of the Datadog tracer, you can override the following defaults:
       tracer: Datadog.tracer,
       debug: false,
       trace_agent_hostname: 'localhost',
-      trace_agent_port: 8126
+      trace_agent_port: 8126,
+      env: Rails.env
```
Comment (robbkidd) [inline docs/GettingStarted.md]: It looks like this change has yielded unexpected results, at least for me. With this change, some spans from the app—and _only some_ but not all, oddly—are getting `env:production` set.

I have a Rails application that we run in multiple named production-like environments, e.g. `staging` and `production`. Both systems' `Rails.env` is `production` per Rails conventions. I differentiate between them by setting the `env:` tag in the agent's `datadog.conf`.

At a minimum, I'd call this change backwards-incompatible: upgrading the gem results in data getting sent with an unexpected `env:` tag.

Was this change to handle deployments where a single Datadog agent on a host would be receiving and forwarding data from multiple applications (not just development/test/production) and so therefore the application needs a way to name its stuff (traces? spans? I'm new here.).

I'm new to APMing with Datadog, so I fully admit I could be doing something wrong or could be doing something better.

File: lib/ddtrace/contrib/rails/framework.rb [hunk 2]
```diff
@@ -26,7 +26,8 @@ module Framework
           tracer: Datadog.tracer,
           debug: false,
           trace_agent_hostname: Datadog::Writer::HOSTNAME,
-          trace_agent_port: Datadog::Writer::PORT
+          trace_agent_port: Datadog::Writer::PORT,
+          env: ::Rails.env
```
Comment (robbkidd) [inline lib/ddtrace/contrib/rails/framework.rb]: It looks like this change has yielded unexpected results, at least for me. With this change, some spans from the app—and _only some_ but not all, oddly—are getting `env:production` set.

I have a Rails application that we run in multiple named production-like environments, e.g. `staging` and `production`. Both systems' `Rails.env` is `production` per Rails conventions. I differentiate between them by setting the `env:` tag in the agent's `datadog.conf`.

At a minimum, I'd call this change backwards-incompatible: upgrading the gem results in data getting sent with an unexpected `env:` tag.

Was this change to handle deployments where a single Datadog agent on a host would be receiving and forwarding data from multiple applications (not just development/test/production) and so therefore the application needs a way to name its stuff (traces? spans? I'm new here.).

What do ya'll think of a more specific default as a combination of the Rails application name and the Rails environment?

```
> Rails.application.class.to_s.split("::").first.downcase + '-' + Rails.env
=> "myrailsapp-production"
```

I'm new to APMing with Datadog, so I fully admit I could be doing something wrong or could be doing something better.

Comment (robbkidd) [inline lib/ddtrace/contrib/rails/framework.rb]: For what it's worth, `env: nil` in my initializer brings back previous behavior (`env:` tag applied by the DD agent config).

Comment (intelekshual) [inline lib/ddtrace/contrib/rails/framework.rb]: In my particular case, yes, I'm using a single Datadog agent per host, with each host receiving and forwarding data from multiple applications (specifically via Docker/Kubernetes). The Rails convention is _not_ to have your `staging` and `production` environments use the same environment name. Instead you should have environments with different names that share configuration.

With that being said, maybe the easiest solution to handle your use-case is to make it possible to set `env:` to `nil` on the Rails side, forcing it to inherit `env` from your agent config.

Comment (robbkidd) [inline lib/ddtrace/contrib/rails/framework.rb]: > The Rails convention is not to have your staging and production environments use the same environment name.

Hunh. Today I Learned ...

OK. I can set aside my philosophical differences with what turns out to be convention and make `nil` my default.  Thanks!

Comment (robbkidd) [inline lib/ddtrace/contrib/rails/framework.rb]: Do you think that `Rails.application.class.to_s.split("::").first.downcase + '-' + Rails.env` as the default is just too silly to look at? (It's pretty silly.)

Comment (ufoot) [inline lib/ddtrace/contrib/rails/framework.rb]: Thanks for all this detailed feedback, FYI, we're trying to address this issue in #106

**Pull request closed without merging**